### PR TITLE
Upgrade CodeStyle and NetCompilersToolset package + fix new code style violations

### PR DESCRIPTION
**Pull request by mavasani**
Recommended to review each commit separately

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Data/ReviewSqlQueriesForSecurityVulnerabilities.cs [hunk 1]
```diff
@@ -94,9 +94,11 @@ public override void Initialize(AnalysisContext context)
                             return;
                         }
 
+                        M();
                         AnalyzeMethodCall(operationContext, invocation.TargetMethod, symbol, invocation.Arguments, invocation.Syntax, isInDbCommandConstructor, isInDataAdapterConstructor, iDbCommandType, iDataAdapterType);
                     }, OperationKind.Invocation);
 
+                    static bool M() => true;
```
Comment (sharwell) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Data/ReviewSqlQueriesForSecurityVulnerabilities.cs]: 😕 The fix all changed this?

Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Data/ReviewSqlQueriesForSecurityVulnerabilities.cs]: No, I was testing something before working on this branch and forgot to revert it :(

Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Data/ReviewSqlQueriesForSecurityVulnerabilities.cs]: Reverted in https://github.com/dotnet/roslyn-analyzers/pull/4086/commits/7b7c30ae5efefa32118df1096a1969f77e4780a0

File: src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb [hunk 2]
```diff
@@ -6,7 +6,9 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 
 Namespace Global.Analyzer.Utilities.Extensions
 
+#Disable Warning IDE1006 ' Naming Styles - False positive (missing prefix 'I')
```
Comment (sharwell) [inline src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb:9]: 😕 Where is this rule coming from? I don't see it in .editorconfig.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/Helpers/InsecureDeserializationTypeDecider.cs [hunk 3]
```diff
@@ -334,9 +334,13 @@ ImmutableArray<InsecureObjectGraphResult> Compute((ITypeSymbol, ObjectGraphOptio
...
-                            || knownTypeAttributeData.ConstructorArguments.Length != 1
-                            || knownTypeAttributeData.ConstructorArguments[0] is not TypedConstant typedConstant
-                            || typedConstant.Kind != TypedConstantKind.Type    // Not handling the string methodName overload
+                            || knownTypeAttributeData.ConstructorArguments.Length != 1)
+                        {
+                            continue;
+                        }
+
+                        TypedConstant typedConstant = knownTypeAttributeData.ConstructorArguments[0];
+                        if (typedConstant.Kind != TypedConstantKind.Type    // Not handling the string methodName overload
```
Comment (sharwell) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/Helpers/InsecureDeserializationTypeDecider.cs:343]: ❔ Can we just go back to the original form here?

File: src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb (hunk 2 above)
Comment (mavasani) [reply src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb:9]: https://github.com/dotnet/roslyn-analyzers/blob/dba3183193df36d666fc48c0eac0d2d640d0041d/.editorconfig#L188

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/Helpers/InsecureDeserializationTypeDecider.cs (hunk 3 above)
Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/Helpers/InsecureDeserializationTypeDecider.cs:343]: Yes, sure I can do that as well. However, seemed like an unnecessary is expression check. @dotpaul What do you prefer?

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypesThatOwnDisposableFieldsShouldBeDisposable.cs [hunk 4]
```diff
@@ -60,7 +60,9 @@ protected abstract class DisposableFieldAnalyzer
         {
             private readonly DisposeAnalysisHelper _disposeAnalysisHelper;
 
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
```
Comment (sharwell) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypesThatOwnDisposableFieldsShouldBeDisposable.cs:63]: ❔ How is this a false positive? Can you not use `RosylnDebug.Assert`?

Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypesThatOwnDisposableFieldsShouldBeDisposable.cs]: Thanks, I was expecting Debug.Assert to take care of it. Changed now.

File: src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb (hunk 2 above)
Comment (Youssef1313) [reply src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb:9]: File bug for it in Roslyn?

Comment (mavasani) [reply src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb:9]: @Youssef1313 bug for what? This is the design - we are explictly requesting all Style analyzers coming from CodeStyle package to be warning diagnostics. This way when we move to a newer package, we automatically get build warnings for newly added analyzers. This basically allows us to switch from opt-in mode to opt-out mode for analyzer warnings from this package.

File: src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs [hunk 5]
```diff
@@ -72,7 +72,7 @@ private static void HandleOperation(in OperationAnalysisContext context, INamedT
         private static void HandleBinaryOperator(in OperationAnalysisContext context, INamedTypeSymbol symbolType)
         {
             var binary = (IBinaryOperation)context.Operation;
-            if (binary.OperatorKind != BinaryOperatorKind.Equals && binary.OperatorKind != BinaryOperatorKind.NotEquals)
+            if (binary.OperatorKind is not BinaryOperatorKind.Equals and not BinaryOperatorKind.NotEquals)
```
Comment (Youssef1313) [inline src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs:75]: Wow! Didn't know that `and not` is an acceptable syntax! 🎉

File: src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb (hunk 2 above)
Comment (Youssef1313) [reply src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb:9]: @mavasani Isn't it false positive? Modules doesn't have to be prefixed with `I` (unlike interfaces).

Comment (mavasani) [reply src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb:9]: @Youssef1313 Ah yes, sorry I misunderstood your suggestion. It was not for the fact that it is escalated to the warning, but the fact that the analyzer itself is reporting a false positive. I'll file a Roslyn bug.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/Helpers/InsecureDeserializationTypeDecider.cs (hunk 3 above)
Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/Helpers/InsecureDeserializationTypeDecider.cs]: I have reverted the changes with a source suppression.

File: src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb (hunk 2 above)
Comment (mavasani) [reply src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb:9]: Alright, so seems like this is not an analyzer issue - I had introduced this rule in Tools Options during my testing and forgot to delete it. I will revert these suppressions.

Comment (mavasani) [reply src/Utilities/Refactoring.VisualBasic/Extensions/StatementSyntaxExtensions.vb]: Reverted now.

**Comment by mavasani** [comment]
@sharwell Any further feedback?

**Comment by mavasani** [comment]
Thanks Sam.

**Pull request merged**